src/alias.cairo

```diff
@@ -0,0 +1 @@
+type ID = felt252;
```

Comment:
commands::uuid() was originally returning a felt252, but now it's returning a usize so i'm thinking that having all IDs being usize as well might be better, wdyt?

Reply:
FWIW, I think felt252 is fine.

I actually don't like Dojo's use of `usize` there for 2 reasons: 1) UUID is defined in a RFC as 128 bits so if we're using the UUID name (even though not the implementation details), it should be `u128` and 2) `usize` is platform dependent, kind of, and I'm not really sure how this will play out with Starknet or other L3s the contracts might be deployed on in the future. It could be aliased to u16 or u256, who knows? It's the uncertainty of it that bothers me.

So I think felt252 or u128 are good options for ID.

Reply:
agreed, let's use u128 then

Reply:
@milancermak I'm hesitating to use type ID for resource_id here because I don't think resource_id is actually an ID but rather a type (not more than 255 resource types so u8 works here). So i'm thinking it might be better to rename as resource_type and keep u8 here.

```
fn has_resource(self: Realm, resource_id: u8) -> bool {
        let mut resource_ids: Span<u8> = unpack_resource_ids(
            self.resource_ids_packed, self.resource_ids_count
        );
```

Reply:
Sound good! I agree with all.